### ISSUE TITLE
4.24.0 turns table names to lowercase - fails to read DATABASECHANGELOG

### ISSUE THREAD

**Issue by fgalarza-macquarie** [liquibase/liquibase#5034]
### Search first

- [X] I searched and no similar issues were found

### Description

**The problem**
Liquibase 4.24.0 now seems to transform table names to lowercase. This breaks everything:
- Cannot read its own `DATABASECHANGELOG` tables (tries to read `databasechangelog` instead).
- This causes it to want to create a new `databasechangelog` table and redeploy all the changesets that have already been deployed.

The problem happens with both Google BigQuery and Spanner. I haven't tried other databases.

**The environment**
I am running `liquibase update-sql` in an Ubuntu:focal container with the Google Spanner and BigQuery add-ons. Versions are as follow:
Liquibase version: 4.24.0
[Spanner add-on](https://github.com/cloudspannerecosystem/liquibase-spanner/releases) version: 4.23.0 
[BigQuery add-on](https://github.com/liquibase/liquibase-bigquery/releases) version: 4.24.0
[BigQuery JDBC driver](https://cloud.google.com/bigquery/docs/reference/odbc-jdbc-drivers) version: 42_1.5.0.1001
Base image: ubuntu:focal

```
$ java --version
openjdk 11.0.20.1 2023-08-24
OpenJDK Runtime Environment (build 11.0.20.1+1-post-Ubuntu-0ubuntu120.04)
OpenJDK 64-Bit Server VM (build 11.0.20.1+1-post-Ubuntu-0ubuntu120.04, mixed mode, sharing)
```

```
$ liquibase --version
[2023-10-09 11:08:45] INFO [liquibase.integration] Starting command execution.
[2023-10-09 11:08:45] INFO [liquibase.integration] Generated version info
####################################################
##   _     _             _ _                      ##
##  | |   (_)           (_) |                     ##
##  | |    _  __ _ _   _ _| |__   __ _ ___  ___   ##
##  | |   | |/ _` | | | | | '_ \ / _` / __|/ _ \  ##
##  | |___| | (_| | |_| | | |_) | (_| \__ \  __/  ##
##  \_____/_|\__, |\__,_|_|_.__/ \__,_|___/\___|  ##
##              | |                               ##
##              |_|                               ##
##                                                ## 
##  Get documentation at docs.liquibase.com       ##
##  Get certified courses at learn.liquibase.com  ## 
##                                                ##
####################################################
Starting Liquibase at 11:08:45 (version 4.24.0 #14062 built at 2023-09-28 12:18+0000)
Liquibase Home: /liquibase
Java Home /usr/lib/jvm/java-11-openjdk-amd64 (Version 11.0.20.1)
Libraries:
- internal/lib/commons-collections4.jar: Apache Commons Collections 4.4.0 By The Apache Software Foundation
- internal/lib/commons-lang3.jar: Apache Commons Lang 3.12.0 By The Apache Software Foundation
- internal/lib/commons-text.jar: Apache Commons Text 1.10.0 By The Apache Software Foundation
- internal/lib/compiler.jar: compiler UNKNOWN
- internal/lib/h2.jar: H2 Database Engine 2.2.222 By H2 Group
- internal/lib/hsqldb.jar: HSQLDB 2.7.2 By The HSQL Development Group
- internal/lib/jaxb-api.jar: jaxb-api 2.3.1 By Oracle Corporation
- internal/lib/jaxb-core.jar: JAXB Core 4.0.3 By Eclipse Foundation
- internal/lib/jaxb-runtime.jar: JAXB Runtime 4.0.3 By Eclipse Foundation
- internal/lib/jaybird.jar: Jaybird 5.0.2.java8 (build: variant=jaybird tag=v5.0.2 date=202306070943) By Firebird project
- internal/lib/jcc.jar: IBM JCC JDBC 4 Driver 1.4.0 By IBM
- internal/lib/liquibase-commercial.jar: Commercial Liquibase Functionality 4.24.0 By Liquibase
- internal/lib/mariadb-java-client.jar: mariadb-java-client 3.2.0 By mariadb.com
- internal/lib/mssql-jdbc.jar: Microsoft JDBC Driver for SQL Server 12.4.1 By Microsoft Corporation
- internal/lib/ojdbc8.jar: JDBC 19.20.0.0.0 By Oracle Corporation
- internal/lib/opencsv.jar: opencsv 5.8.0
- internal/lib/picocli.jar: picocli 4.7.5 By Remko Popma
- internal/lib/postgresql.jar: PostgreSQL JDBC Driver 42.6.0 By PostgreSQL Global Development Group
- internal/lib/snakeyaml.jar: SnakeYAML 2.2.0
- internal/lib/snowflake-jdbc.jar: snowflake-jdbc 3.14.1
- internal/lib/sqlite-jdbc.jar: SQLite JDBC 3.43.0.0
- lib/GoogleBigQueryJDBC42.jar: SimbaJDBCDriverforGoogleBigQuery SimbaJDBCDriverforGoogleBigQuery_1.5.0.1001 By Simba Technologies Inc.
- lib/animal-sniffer-annotations-1.23.jar: animal-sniffer-annotations-1.23 UNKNOWN
- lib/annotations-4.1.1.4.jar: annotations-4.1.1.4 UNKNOWN
- lib/api-common-2.15.0.jar: API Common 2.15.0 By Google LLC
- lib/auto-value-1.10.2.jar: auto-value-1.10.2 UNKNOWN
- lib/auto-value-annotations-1.10.2.jar: auto-value-annotations-1.10.2 UNKNOWN
- lib/avro-1.11.2.jar: Apache Avro 1.11.2 By The Apache Software Foundation
- lib/commons-codec-1.15.jar: Apache Commons Codec 1.15.0 By The Apache Software Foundation
- lib/commons-compress-1.22.jar: Apache Commons Compress 1.22.0 By The Apache Software Foundation
- lib/commons-logging-1.2.jar: Apache Commons Logging 1.2.0 By The Apache Software Foundation
- lib/conscrypt-openjdk-uber-2.5.2.jar: org.conscrypt 2.5.2
- lib/error_prone_annotations-2.18.0.jar: error_prone_annotations-2.18.0 UNKNOWN
- lib/failureaccess-1.0.1.jar: Guava InternalFutureFailureAccess and InternalFutures 1.0.1
- lib/gax-2.32.0.jar: GAX (Google Api eXtensions) for Java (Core) 2.32.0 By Google LLC
- lib/gax-grpc-2.32.0.jar: GAX (Google Api eXtensions) for Java (gRPC) 2.32.0 By Google LLC
- lib/google-api-client-2.2.0.jar: Google APIs Client Library for Java 2.2.0 By Google
- lib/google-api-services-bigquery-v2-rev20230805-2.0.0.jar: google-api-services-bigquery-v2-rev20230805-2.0.0 UNKNOWN
- lib/google-auth-library-credentials-1.19.0.jar: google-auth-library-credentials-1.19.0 UNKNOWN
- lib/google-auth-library-oauth2-http-1.19.0.jar: google-auth-library-oauth2-http-1.19.0 UNKNOWN
- lib/google-cloud-bigquerystorage-2.41.1.jar: BigQuery Storage 2.41.1 By Google LLC
- lib/google-cloud-core-2.22.0.jar: Google Cloud Core 2.22.0 By Google LLC
- lib/google-http-client-1.43.3.jar: Google HTTP Client Library for Java 1.43.3 By Google
- lib/google-http-client-apache-v2-1.42.3.jar: Apache HTTP transport v2 for the Google HTTP Client Library for Java. 1.42.3 By Google
- lib/google-http-client-gson-1.43.3.jar: google-http-client-gson-1.43.3 UNKNOWN
- lib/google-oauth-client-1.34.1.jar: Google OAuth Client Library for Java 1.34.1 By Google
- lib/grpc-context-1.56.1.jar: jar 1.56.1
- lib/grpc-google-cloud-bigquerystorage-v1-2.41.1.jar: grpc-google-cloud-bigquerystorage-v1 2.41.1 By Google LLC
- lib/grpc-google-cloud-bigquerystorage-v1beta1-0.165.1.jar: grpc-google-cloud-bigquerystorage-v1beta1 0.165.1 By Google LLC
- lib/grpc-google-cloud-bigquerystorage-v1beta2-0.165.1.jar: grpc-google-cloud-bigquerystorage-v1beta2 0.165.1 By Google LLC
- lib/gson-2.10.1.jar: Gson 2.10.1 By Google Gson Project
- lib/guava-32.1.2-jre.jar: Guava: Google Core Libraries for Java 32.1.2.jre
- lib/httpclient-4.5.14.jar: Apache HttpClient 4.5.14 By The Apache Software Foundation
- lib/httpcore-4.4.16.jar: HttpComponents Apache HttpCore 4.4.16 By The Apache Software Foundation
- lib/j2objc-annotations-2.8.jar: j2objc-annotations-2.8 UNKNOWN
- lib/jackson-annotations-2.14.2.jar: Jackson-annotations 2.14.2 By FasterXML
- lib/jackson-core-2.14.2.jar: Jackson-core 2.14.2 By FasterXML
- lib/jackson-databind-2.14.2.jar: jackson-databind 2.14.2 By FasterXML
- lib/javax.annotation-api-1.3.2.jar: javax.annotation API 1.3.2 By GlassFish Community
- lib/joda-time-2.10.10.jar: Joda-Time 2.10.10 By Joda.org
- lib/json-20230618.jar: JSON in Java 20230618.0.0
- lib/jsr305-3.0.2.jar: FindBugs-jsr305 3.0.2
- lib/liquibase-bigquery-4.24.0.jar: liquibase-bigquery-4.24.0 UNKNOWN
- lib/liquibase-spanner-4.23.0-all.jar: liquibase-spanner-4.23.0-all UNKNOWN
- lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar: listenablefuture-9999.0-empty-to-avoid-conflict-with-guava UNKNOWN
- lib/opencensus-api-0.31.1.jar: opencensus-api 0.31.1
- lib/opencensus-contrib-http-util-0.31.1.jar: opencensus-contrib-http-util 0.31.1
- lib/perfmark-api-0.26.0.jar: PerfMark 0.26.0 By Carl Mastrangelo
- lib/proto-google-cloud-bigquerystorage-v1-2.41.1.jar: proto-google-cloud-bigquerystorage-v1 2.41.1 By Google LLC
- lib/proto-google-cloud-bigquerystorage-v1beta1-0.165.1.jar: proto-google-cloud-bigquerystorage-v1beta1 0.165.1 By Google LLC
- lib/proto-google-cloud-bigquerystorage-v1beta2-0.165.1.jar: proto-google-cloud-bigquerystorage-v1beta2 0.165.1 By Google LLC
- lib/proto-google-common-protos-2.23.0.jar: proto-google-common-protos 2.23.0 By Google LLC
- lib/proto-google-iam-v1-1.18.0.jar: proto-google-iam-v1 1.18.0 By Google LLC
- lib/protobuf-java-3.23.2.jar: Protocol Buffers [Core] 3.23.2
- lib/protobuf-java-util-3.23.2.jar: Protocol Buffers [Util] 3.23.2
- lib/slf4j-api-1.7.36.jar: slf4j-api 1.7.36 By SLF4J.ORG
- lib/threetenbp-1.6.8.jar: ThreeTen backport 1.6.8 By ThreeTen.org


Liquibase Version: 4.24.0
Liquibase Open Source 4.24.0 by Liquibase
```

### Steps To Reproduce

Create dummy SQL changelog file:
```
--liquibase formatted sql logicalFilePath:foo/bar/000.sql

-- changeset Fran:1
CREATE TABLE table_0000 (
    name STRING
);

-- changeset Fran:2
CREATE TABLE TABLE_0002 (
    name STRING
);

-- changeset Fran:3
CREATE TABLE TaBlE_0003 (
    name STRING
);
```

Create a `liquibase.bigquery.properties` file
```
driver:com.simba.googlebigquery.jdbc.Driver
logLevel: INFO
```

Execute a `liquibase update-sql` command against an existing dataset
```
PROJECT_ID="<your GCP project ID>"
DATASET="<your bigquery dataset>"
LOCATION="<your bigquery location>"

liquibase update-sql \
    --defaultsFile="liquibase.bigquery.properties" \
    --changelog-file="000.sql" \
    --liquibaseSchemaName="$DATASET" \
    --url="jdbc:bigquery://https://www.googleapis.com/bigquery/v2:443;ProjectId=$PROJECT_ID;DefaultDataset=$DATASET;OAuthType=3;Location=$LOCATION;"
```

Observe how the output tries to recreate the `DATABASECHANGELOG` and `DATABASECHANGELOGLOCK` in lowercase:
```
-- Create Database Lock Table
CREATE TABLE test.databasechangeloglock (ID INT, LOCKED BOOLEAN, LOCKGRANTED datetime, LOCKEDBY STRING(255));

-- Initialize Database Lock Table
DELETE FROM test.databasechangeloglock WHERE 1 = 1;

INSERT INTO test.databasechangeloglock (ID, LOCKED) VALUES (1, false);

-- Lock Database
UPDATE test.databasechangeloglock SET LOCKED = true, LOCKEDBY = '1ccf1acd4a0a (172.17.0.2)', LOCKGRANTED = CURRENT_DATETIME() WHERE ID = 1 AND LOCKED = false;

-- Create Database Change Log Table
CREATE TABLE test.databasechangelog (ID STRING(255), AUTHOR STRING(255), FILENAME STRING(255), DATEEXECUTED datetime, ORDEREXECUTED INT, EXECTYPE STRING(10), MD5SUM STRING(35), DESCRIPTION STRING(255), COMMENTS STRING(255), TAG STRING(255), LIQUIBASE STRING(20), CONTEXTS STRING(255), LABELS STRING(255), DEPLOYMENT_ID STRING(10));

-- *********************************************************************
-- Update Database Script
-- *********************************************************************
-- Change Log: 000.sql
-- Ran at: 10/9/23, 10:54 AM
-- Against: @jdbc:bigquery://https://www.googleapis.com/bigquery/v2:443;ProjectId=<REDACTED_PROJECT_ID>;DefaultDataset=ai;OAuthType=3;Location=<REDACTED_LOCATION>;
-- Liquibase version: 4.24.0
-- *********************************************************************

-- Changeset foo/bar/000.sql::1::Fran
CREATE TABLE table_0000 (
    name STRING
);

INSERT INTO test.databasechangelog (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('1', 'Fran', 'foo/bar/000.sql', CURRENT_DATETIME(), 1, '9:2a898c7e87f43c35f48e658aa9d64763', 'sql', '', 'EXECUTED', NULL, NULL, '4.24.0', NULL);

-- Changeset foo/bar/000.sql::2::Fran
CREATE TABLE TABLE_0002 (
    name STRING
);

INSERT INTO test.databasechangelog (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('2', 'Fran', 'foo/bar/000.sql', CURRENT_DATETIME(), 2, '9:2fb6ca1398676b82c094aa5615e80d09', 'sql', '', 'EXECUTED', NULL, NULL, '4.24.0', NULL);

-- Changeset foo/bar/000.sql::3::Fran
CREATE TABLE TaBlE_0003 (
    name STRING
);

INSERT INTO test.databasechangelog (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('3', 'Fran', 'foo/bar/000.sql', CURRENT_DATETIME(), 3, '9:88bb929ae06096ac06c8169882d6f63f', 'sql', '', 'EXECUTED', NULL, NULL, '4.24.0', NULL);

-- Release Database Lock
UPDATE test.databasechangeloglock SET LOCKED = false, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1;
```

### Expected/Desired Behavior

- Liquibase recognises the existing `DATABASECHANGELOG` and `DATABASECHANGELOGLOCK` tables (in upper case) and does not try to recreate them.
- Liquibase reads the changesets that have already been applied from `DATABASECHANGELOG` and does not try to apply them again.

### Liquibase Version

4.24.0

### Database Vendor & Version

BigQuery

### Liquibase Integration

CLI

### Liquibase Extensions

liquibase-spanner-4.23.0, liquibase-bigquery-4.24.0

### OS and/or Infrastructure Type/Provider

Ubuntu

### Additional Context

This worked just fine with `4.23.2`.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by tati-qalified** [comment]
Hi @fgalarza-macquarie, thank you for bringing this to our attention. I've been able to verify that this is an issue, and we're evaluating how to move forward.
Thank you,
Tatiana

**Comment by tati-qalified** [comment]
@filipelautert it doesn't, it uses the name but not the case. It will create them as lowercase anyway

**Comment by filipelautert** [comment]
@tati-qalified @fgalarza-macquarie added a fix to liquibase-core with new apis to fix this problem. Now we need to release liquibase 4.26.0 and then bigquery PR https://github.com/liquibase/liquibase-bigquery/pull/243 can be fully tested and merged prior to releasing bigquery extension 4.26.0 .
